concourse/templates/default.mako

```diff
@@ -213,6 +213,70 @@ if secret_cfg:
 % endif
 </%def>
 
+<%def name="on_abort(indent, job_variant, job_step)" filter="indent_func(indent),trim">
+<%
+if job_variant.has_main_repository():
+  source_repo = job_variant.main_repository()
+  source_repo_github_cfg_name = source_repo.cfg_name() or github.name()
+else:
+  source_repo = None
+  source_repo_github_cfg_name = github.name()
+%>
+on_abort:
```

Comment:
can this be deduplicated from the existing step-code? I assume the only difference should be indentation level + `on_abort` toplevel attr, right?